posthog/hogql/expr_parser.py

```diff
@@ -0,0 +1,187 @@
+import ast
+import re
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from clickhouse_driver.util.escape import escape_param
+
+from posthog.models.property.util import get_property_string_expr
+
+EVENT_FIELDS = ["id", "uuid", "event", "timestamp", "distinct_id"]
+PERSON_FIELDS = ["id", "created_at", "properties"]
+CLICKHOUSE_FUNCTIONS = ["concat", "coalesce"]
+HOGQL_AGGREGATIONS = ["avg", "sum", "total"]
+
+
+@dataclass
+class ExprParserContext:
+    property_list: List[List[str]] = field(default_factory=list)
```

Comment:
This overloads the term "property", since only a subset of values of this list are actually properties, e.g. `timestamp` is not. Maybe `attribute_list` (kind of following Postgres parlance)?